### PR TITLE
Adds new X-Cache-Info header to the xdebug plugin

### PR DESCRIPTION
**Pull request by elsloo**
* Adds new X-Cache-Info header to the xdebug plugin
* Injects the `X-Cache-Info` header with value of `path=/path/to/disk/from/storage/config; volume=n` where `n` is the volume number
* Fixes a small bug inside `HttpCacheSM.h` in `get_volume_number()` that leads to `-1` being returned on cache miss
  * NOTE: this will change behavior of any plugin that relies on `TSHttpTxnInfoIntGet` with a key of `TS_TXN_INFO_CACHE_VOLUME`
* NOTE: This PR depends on #7783

### CODE REVIEW

**Comment by ywkaras** [comment]
Could you add an Au test or add to x_remap.test.py?

**Comment by elsloo** [comment]
> Could you add an Au test or add to x_remap.test.py?

Done. I added a separate `x_cach_info` Au test that's mostly a copy of `x_remap`.

**Comment by elsloo** [comment]
[approve ci]

File: tests/gold_tests/pluginTest/xdebug/x_cache_info/x_cache_info.test.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+tr.Processes.Default.Command = "cp {}/tcp_client.py {}/tcp_client.py".format(
+    Test.Variables.AtsTestToolsDir, Test.RunDirectory)
+tr.Processes.Default.ReturnCode = 0
+
+
+def sendMsg(msgFile):
+
+    tr = Test.AddTestRun()
+    tr.Processes.Default.Command = (
+        "( python {}/tcp_client.py 127.0.0.1 {} {}/{}.in".format(
```
Comment (randall) [inline tests/gold_tests/pluginTest/xdebug/x_cache_info/x_cache_info.test.py:60]: does this need to be python3? (like x_remap.test.py)?\

**Comment by shinrich** [comment]
Looks like the autest is failing because the gold file includes the sandbox path in the output.

File: tests/gold_tests/pluginTest/xdebug/x_cache_info/x_cache_info.test.py (hunk 1 above)
Comment (bryancall) [reply tests/gold_tests/pluginTest/xdebug/x_cache_info/x_cache_info.test.py:60]: Yes, we should use python3.

**Review by dragon512 — APPROVED** [review]
Based on the autest..

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Pull request merged**